### PR TITLE
SEMI-1329 - Added unit tests for "print" call and changed test complete condition to print status = done

### PR DESCRIPTION
**Pull request by chrissauerwein**
@mike-clover-com @david-clover-com @clover/clover-semi-int

### CODE REVIEW

File: public/ts/tests/TestImageUrlPrint.ts [hunk 1]
```diff
@@ -67,9 +67,16 @@ export namespace TestImageUrlPrint {
              The connector is ready
              */
             this.displayMessage({message: "Sending print request"});
-            this.cloverConnector.printImageFromURL(window.location.origin + this.imageLocation);
-            setTimeout(function(){this.testComplete(true);}.bind(this), 20000);
-            // this.testComplete(true);
+            this.cloverConnector.printImageFromURL("https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public" + this.imageLocation);
```
Comment (mike-clover-com) [inline public/ts/tests/TestImageUrlPrint.ts]: I *think* I would prefer this reference the window.location, which would decouple this from requiring access to github.  Is there a reason it was changed?

File: public/ts/tests/TestPrint_All_1.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_All_1 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, ["/images/test_receipt_1.jpg"], ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.bmp"], ["Printing one of each thing..."]);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_All_1.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_01.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_01 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/test_receipt_1.jpg"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_01.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_1.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_1 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.bmp"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_1.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_2.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_2 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.gif"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_2.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_3.ts [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_3 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.jpg"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_3.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_4.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_4 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.pdf"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_4.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_5.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_5 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/print_test_0.png"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_5.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_6.ts [hunk 9]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_6 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/test_receipt_7.jpg"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_6.ts]: Same comment as above - want to remove reference to github.

File: public/ts/tests/TestPrint_Url_8.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import {TestPrint} from './TestPrint';
+import {CloverConfigLoader} from '../configurationLoader/CloverConfigLoader';
+
+export class TestPrint_Url_8 extends TestPrint {
+
+    constructor(loader: CloverConfigLoader, progressInfoCallback: any) {
+        // additional params are: images [], urls [], text []
+
+        super(loader, progressInfoCallback, null, ["https://raw.githubusercontent.com/clover/clover-cloud-connector-unit-examples/master/public/images/test_receipt_8.jpg"], null);
```
Comment (mike-clover-com) [inline public/ts/tests/TestPrint_Url_8.ts]: Same comment as above - want to remove reference to github.

**Comment by chrissauerwein** [comment]
Per conversation with Mike, need to leave github references and add some that use window.location references...will do that in a future commit...

**Pull request merged**